Tests/AsyncHTTPClientTests/HTTPClientInternalTests.swift

```diff
@@ -290,7 +290,7 @@ class HTTPClientInternalTests: XCTestCase {
         let httpBin = HTTPBin()
         let httpClient = HTTPClient(eventLoopGroupProvider: .shared(self.clientGroup))
         defer {
-            XCTAssertNoThrow(try httpClient.syncShutdown(requiresCleanClose: true))
+            XCTAssertNoThrow(try httpClient.syncShutdown())
```

Comment:
In the new implementation we can not make guarantees about the order of succeeding/failing the request (which will lead to a call to `syncShutdown`) and marking the connection as idle. For this reason we can not enforce a clean shutdown here.